packages/smooth_app/lib/pages/product/common/product_list_item_simple.dart

```diff
@@ -1,25 +1,74 @@
 import 'package:flutter/material.dart';
-import 'package:openfoodfacts/model/Product.dart';
+import 'package:provider/provider.dart';
 import 'package:smooth_app/cards/product_cards/smooth_product_card_found.dart';
+import 'package:smooth_app/cards/product_cards/smooth_product_card_template.dart';
 import 'package:smooth_app/data_models/product_list.dart';
+import 'package:smooth_app/data_models/up_to_date_product_provider.dart';
+import 'package:smooth_app/database/local_database.dart';
+import 'package:smooth_app/pages/product/common/product_model.dart';
 
 /// Widget for a [ProductList] item (simple product list)
-class ProductListItemSimple extends StatelessWidget {
+class ProductListItemSimple extends StatefulWidget {
   const ProductListItemSimple({
-    required this.product,
+    required this.barcode,
     this.onTap,
     this.onLongPress,
+    this.backgroundColor,
   });
 
-  final Product product;
+  final String barcode;
   final VoidCallback? onTap;
   final VoidCallback? onLongPress;
+  final Color? backgroundColor;
 
   @override
-  Widget build(BuildContext context) => SmoothProductCardFound(
-        heroTag: product.barcode!,
-        product: product,
-        onTap: onTap,
-        onLongPress: onLongPress,
+  State<ProductListItemSimple> createState() => _ProductListItemSimpleState();
+}
+
+class _ProductListItemSimpleState extends State<ProductListItemSimple> {
+  late final ProductModel _model;
+
+  @override
+  void initState() {
+    super.initState();
+    _model = ProductModel(
+      widget.barcode,
+      context.read<LocalDatabase>(),
+    );
+  }
+
+  @override
+  Widget build(BuildContext context) => Consumer<UpToDateProductProvider>(
+        builder: (
+          final BuildContext context,
+          final UpToDateProductProvider provider,
+          final Widget? child,
+        ) =>
+            ChangeNotifierProvider<ProductModel>(
+          create: (final BuildContext context) => _model,
+          builder: (final BuildContext context, final Widget? wtf) {
+            context.watch<ProductModel>();
+            _model.setRefreshedProduct(provider.getFromBarcode(widget.barcode));
+            switch (_model.loadingStatus) {
+              case LoadingStatus.LOADING:
+                return const SmoothProductCardTemplate();
+              case LoadingStatus.ERROR:
+                return Text('Fatal Error: ${_model.loadingError}');
+              case LoadingStatus.LOADED:
+            }
+            if (_model.product == null) {
+              // TODO(monsieurtanuki): button "Something strange happened: would you like to download that product again?"
+              return Text(
+                  'Oops: could not not this product (${widget.barcode})');
```

Comment:
```suggestion
                  'Oops: could not load this product (${widget.barcode})');
```

Also translation would be good but not a priority here

Reply:
No it's definitely not a priority, as there's a more relevant pending `TODO: button "Something strange happened: would you like to download that product again?"`